source/microbit/main.cpp

```diff
@@ -30,3 +30,37 @@ void app_main() {
         mp_run();
     }
 }
+
+static void ticker(void) {
+
+    // increment our real-time counter.
+    ticks += FIBER_TICK_PERIOD_MS;
+
+    /** Update compass if it is calibrating, but not if it is still
+     *  updating as compass.idleTick() is not reentrant.
+     */
+    if (uBit.compass.isCalibrating() && !compass_updating) {
+        uBit.compass.idleTick();
+    }
+
+    compass_up_to_date = false;
+    accelerometer_up_to_date = false;
+
+    // Update buttons
+    uBit.buttonA.systemTick();
+    uBit.buttonB.systemTick();
+
+    // Update the display.
+    microbit_display_tick();
+}
+
+extern "C" {
+
+void microbit_init(void) {
+    microbit_display_init();
+
+    // Hijack the DAL system ticker.
+    uBit.systemTicker.attach(ticker, MICROBIT_DISPLAY_REFRESH_PERIOD);
```

Comment:
With this PR we no longer calling the DAL's `systemTick`. 
The entry points to the DAL fiber scheduler were `systemTick()`, fiber_sleep()` and `schedule()`. This PR drops all references to those functions, ensuring that MicroPython, and not the DAL, now controls what runs and when.